github/resource_github_organization_ruleset.go

```diff
@@ -713,3 +714,50 @@ func resourceGithubOrganizationRulesetImport(d *schema.ResourceData, meta any) (
 
 	return []*schema.ResourceData{d}, nil
 }
+
+func validateConditionsFieldForBranchAndTagTargets(d *schema.ResourceDiff, meta interface{}) error {
+	conditions := d.Get("conditions").([]any)[0].(map[string]any)
+	if conditions["ref_name"] == nil || conditions["repository_name"] == nil || conditions["repository_id"] == nil {
+		return fmt.Errorf("ref_name and repository_name or ref_name and repository_id must be set for branch and tag targets")
+	}
```

Comment:
This should be simplified to make what is required a bit more clear.

```suggestion
    if conditions["ref_name"] == nil {
        return fmt.Errorf("ref_name must be set for branch and tag targets")
    }
    if conditions["repository_name"] == nil && conditions["repository_id"] == nil {
        return fmt.Errorf("repository_name or repository_id must be set for branch and tag targets")
    }
```